src/Abstractions/src/Asp.Versioning.Abstractions/DeprecationPolicy.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+
+namespace Asp.Versioning;
+
+/// <summary>
+/// Represents an API version deprecation policy.
+/// </summary>
+public class DeprecationPolicy
+{
+    private readonly LinkList links;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DeprecationPolicy"/> class.
+    /// </summary>
+    public DeprecationPolicy()
+    {
+        links = new DeprecationLinkList();
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DeprecationPolicy"/> class.
+    /// </summary>
+    /// <param name="date">The date and time when the API version will be deprecated.</param>
+    /// <param name="link">The optional link which provides information about the deprecation policy.</param>
+    public DeprecationPolicy( DateTimeOffset date, LinkHeaderValue? link = default )
+        : this()
+    {
+        Date = date;
+
+        if ( link is not null )
+        {
+            links.Add( link );
+        }
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="DeprecationPolicy"/> class.
+    /// </summary>
+    /// <param name="link">The link which provides information about the deprecation policy.</param>
+    public DeprecationPolicy( LinkHeaderValue link )
```

Comment:
nit: this can be:

```c#
public DeprecationPolicy( LinkHeaderValue link ) => links.Add( link );
```

>We're covered by the `null`  check because `Collection<T>` does not allow adding or updating with `null`